### PR TITLE
store, tikv: let client of mock-tikv support dag req.

### PR DESCRIPTION
**Pull request by hanfei1991**
left tikv and mock-tikv's cop client behave differently.

@shenli @coocood @zimulala PTAL

### CODE REVIEW

File: store/tikv/store_test.go [hunk 1]
```diff
@@ -38,14 +38,9 @@ type testStoreSuite struct {
...
 func (s *testStoreSuite) SetUpTest(c *C) {
-	s.cluster = mocktikv.NewCluster()
-	mocktikv.BootstrapWithSingleStore(s.cluster)
-	mvccStore := mocktikv.NewMvccStore()
-	clientFactory := mocktikv.NewRPCClient(s.cluster, mvccStore)
-	pdCli := &codecPDClient{mocktikv.NewPDClient(s.cluster)}
-	store, err := newTikvStore("mock-tikv-store", pdCli, clientFactory, false)
-	c.Assert(err, IsNil)
-	s.store = store
+	store, err := NewMockTikvStore("")
```
Comment (zimulala) [inline store/tikv/store_test.go]: Why not use `NewMockTikvStoreWithCluster`?
If use this one, remove the line34.

Comment (hanfei1991) [inline store/tikv/store_test.go]: I'm afraid to change this test ... not very familiar

Comment (hanfei1991) [inline store/tikv/store_test.go]: Well, you are right...

**Comment by zimulala** [comment]
LGTM

File: store/tikv/kv.go [hunk 2]
```diff
@@ -105,9 +105,10 @@ type tikvStore struct {
 	lockResolver *LockResolver
 	gcWorker     *GCWorker
 	etcdAddrs    []string
+	mock         bool
 }
 
-func newTikvStore(uuid string, pdClient pd.Client, client Client, enableGC bool) (*tikvStore, error) {
+func newTikvStore(uuid string, pdClient pd.Client, client Client, enableGC bool, mock bool) (*tikvStore, error) {
```
Comment (coocood) [inline store/tikv/kv.go]: We don't need to pass `mock`, it can be inferred by asserting the type of the client.

Comment (coocood) [inline store/tikv/kv.go]: This function has too many arguments already.

Comment (hanfei1991) [inline store/tikv/kv.go]: ok

**Comment by hanfei1991** [comment]
@coocood PTAL

**Comment by coocood** [comment]
LGTM

**Pull request merged**